server/src/main/java/org/elasticsearch/search/builder/SearchQueryWrapperBuilder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.builder;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.Rewriteable;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.search.query.SearchQueryWrapper;
+import org.elasticsearch.usage.SearchUsage;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * {@code SearchQueryBuilder} is a wrapper class for containing all
+ * the information required to perform a single search query
+ * as part of a series of multiple queries for features like ranking.
+ * It's expected to typically be used as part of a {@link java.util.List}.
+ */
+public class SearchQueryWrapperBuilder implements ToXContent, Writeable, Rewriteable<SearchQueryWrapperBuilder> {
```

Comment:
do we want to consider including rank in their name (also in the name of the section) given they only make sense when a rank section is provided?

Reply:
I thought about adding `rank`, but I decided against that because it would be inconsistent with `queries` being a top-level element. Maybe something like `LimitedSearchSourceBuilder`, `ChildSearchSourceBuilder`, or `SubSearchSourceBuilder`?

Reply:
SubSearchSourceBuilder sounds good to me.

Reply:
Otherwise, wild idea: how about we only write a new parser that parses a SearchSourceBuilder?